app/views/questionnaire.py

```diff
@@ -121,6 +121,10 @@ def post_block(eq_id, form_type, collection_id, group_id, group_instance, block_
     else:
         _update_questionnaire_store(current_location, form)
         next_location = path_finder.get_next_location(current_location=current_location)
+
+        if next_location is None:
```

Comment:
I think we should check explicitly for a user action e.g. `if 'action[submit_answers]' in request.form` rather than rely on next_location being None.